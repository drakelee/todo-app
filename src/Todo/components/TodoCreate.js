import React, {Component} from 'react'
import Button from '@material-ui/core/Button/Button'
import Input from '@material-ui/core/Input/Input'

class TodoCreate extends Component {
    state = {
        title: ''
    }

    render() {
        const {title} = this.state
        return (
            <form>
                <Input
                    id='title'
                    placeholder='Todo Title'
                    onChange={this.handleChange('title')}
                    value={title}
                />
                <Button
                    onClick={this.handleClick}
                >
                    <div>Add Todo</div>
                </Button>
            </form>
        )
    }

    handleChange = field => event => {
        this.setState({
            ...this.state,
            [field]: event.target.value
        })
    }

    handleClick = () => {
        alert(this.state.title)
    }
}

export default TodoCreate