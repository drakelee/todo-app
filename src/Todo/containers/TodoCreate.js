import React, {Component} from 'react'
import Button from '@material-ui/core/es/Button/Button'
import Input from '@material-ui/core/es/Input/Input'

class TodoCreate extends Component {
    state = {
        title: ''
    }

    render() {
        const {title} = this.state
        return (
            <form>
                <Input
                    id='title'
                    placeholder='Todo Title'
                    onChange={this.handleChange('title')}
                    value={title}
                />
                <Button
                    onClick={this.handleClick}
                >
                    <div>Add Todo</div>
                </Button>
            </form>
        )
    }

    handleChange = field => event => {
        this.setState({
            ...this.state,
            [field]: event.target.value
        })
    }

    handleClick = () => {
        alert(this.state.title)
    }
}

export default TodoCreate