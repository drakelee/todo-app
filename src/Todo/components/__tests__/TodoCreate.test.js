import React from 'react'
import {createShallow} from '@material-ui/core/test-utils'
import TodoCreate from '../TodoCreate'
import Input from '@material-ui/core/Input/Input'
import renderer from 'react-test-renderer'

describe('<TodoCreate/>', () => {
    const shallow = createShallow()

    it('should change titles with user input', () => {
        const wrapper = shallow(<TodoCreate/>)
        wrapper.find(Input).simulate('change', {
            target: {
                value: 'updated title'
            }
        })
        expect(wrapper.find(Input).props().value).toEqual('updated title')
    })

    it('renders correctly', () => {
        const tree = renderer.create(
            <TodoCreate/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
