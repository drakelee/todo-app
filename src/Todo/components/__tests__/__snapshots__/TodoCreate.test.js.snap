// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoCreate/> renders correctly 1`] = `
<form>
  <div
    className="MuiInput-root-1 MuiInput-underline-5"
  >
    <input
      aria-invalid={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-9"
      defaultValue={undefined}
      disabled={undefined}
      id="title"
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Todo Title"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root-38 MuiButton-root-14 MuiButton-text-16 MuiButton-flat-19"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-15"
    >
      <div>
        Add Todo
      </div>
    </span>
    <span
      className="MuiTouchRipple-root-41"
    />
  </button>
</form>
`;
